src/main/java/org/odessajavaclub/topic/adapter/in/web/TopicController.java

```diff
@@ -0,0 +1,168 @@
+import org.odessajavaclub.topic.application.port.in.UpdateTopicUseCase;
+import org.odessajavaclub.topic.domain.Topic;
+import org.odessajavaclub.topic.domain.enumeration.TopicStatus;
+import org.odessajavaclub.topic.domain.enumeration.TopicType;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.mvc.method.annotation.MvcUriComponentsBuilder;
+
+import javax.validation.Valid;
+import java.net.URI;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/topics")
+public class TopicController {
+  private final CreateTopicUseCase createTopicUseCase;
+  private final UpdateTopicUseCase updateTopicUseCase;
+  private final DeleteTopicUseCase deleteTopicUseCase;
+  private final GetTopicsQuery topicsQuery;
+  private final TopicDtoMapper topicDtoMapper;
+
+  @GetMapping
+  List<TopicDto> listTopics(
+      @RequestParam(value = "page", defaultValue = "0") int page,
+      @RequestParam(value = "size", defaultValue = "100") int size,
+      @RequestParam(value = "sortBy", defaultValue = "event") String sortBy,
+      @RequestParam(value = "order", defaultValue = "ASC") String order) {
+    return topicsQuery.getTopics(sortBy, order, page, size).stream()
+        .map(topicDtoMapper::toGetTopicDto)
+        .collect(Collectors.toList());
+  }
+
+  @GetMapping("/{id}")
+  ResponseEntity<TopicDto> getTopic(@PathVariable Long id) {
+    return topicsQuery
+        .getTopic(new Topic.TopicId(id))
+        .map(topicDtoMapper::toGetTopicDto)
+        .map(ResponseEntity::ok)
+        .orElseGet(() -> ResponseEntity.notFound().build());
+  }
+
+  @GetMapping("/status/{status}")
+  List<TopicDto> listTopicsByAuthor(
+      @PathVariable TopicStatus status,
+      @RequestParam(required = false, defaultValue = "0") int page,
+      @RequestParam(required = false, defaultValue = "100") int size,
+      @RequestParam(value = "sortBy", defaultValue = "event") String sortBy,
+      @RequestParam(value = "order", defaultValue = "ASC") String order) {
+    return topicsQuery.getTopicsByStatus(status, sortBy, order, page, size).stream()
+        .map(topicDtoMapper::toGetTopicDto)
+        .collect(Collectors.toList());
+  }
+
+  @GetMapping("/type/{type}")
+  List<TopicDto> listTopicsByAuthor(
```

Comment:
As we discussed, Slices should be returned instead of lists

Reply:
A lot of refactoring, I need to think about it.